### PR TITLE
Build local image and load to kind cluster

### PR DESCRIPTION
**Pull request by AvaStancu**

### CODE REVIEW

File: .github/workflows/e2e-test-linux-vm.yaml [hunk 1]
```diff
@@ -9,16 +9,33 @@ on:
 
 env:
   TARGET_ORG: actions-runner-controller
+  CLUSTER_NAME: e2e-test
+  RUNNER_VERSION: 2.302.1
```
Comment (Link-) [inline .github/workflows/e2e-test-linux-vm.yaml:13]: What happens when we release a new runner version?

Comment (Link-) [reply .github/workflows/e2e-test-linux-vm.yaml:13]: I think it'll be good to make this an input parameter that is optional and if not provided, we read the version from this file: https://github.com/actions/actions-runner-controller/blob/master/runner/VERSION 

We will always update this file with every new runner version release, it's used to publish the new runner images

File: .github/workflows/e2e-test-linux-vm.yaml [hunk 2]
```diff
@@ -9,16 +9,33 @@ on:
...
+            TAG=$(echo "0.0.$GITHUB_SHA")
+            echo "TAG=$TAG" >> $GITHUB_ENV
+            echo "IMAGE=$(echo "$IMAGE_REPO:$TAG)" >> $GITHUB_ENV
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          version: latest
+      - name: Docker Build Test Image
+        run: |
+            docker buildx build --build-arg RUNNER_VERSION=$RUNNER_VERSION --build-arg TAG=$TAG -t $IMAGE . --load
```
Comment (Link-) [inline .github/workflows/e2e-test-linux-vm.yaml]: Do we need these flags?
```
	export DOCKER_CLI_EXPERIMENTAL=enabled
	export DOCKER_BUILDKIT=1
```
Source: https://github.com/actions/actions-runner-controller/blob/a462ecbe7956f1bd4edd838cdf476dba35b49667/Makefile#L202

File: .github/workflows/e2e-test-linux-vm.yaml (hunk 1 above)
Comment (TingluoHuang) [reply .github/workflows/e2e-test-linux-vm.yaml:13]: will [.github/workflows/update-runners.yaml](https://github.com/actions/actions-runner-controller/pull/2378/files#diff-f88bf13f3a64d7d08ec66bd4fd5d0967dab39a858f1b2a3f16af72324f57198e) take care of runner update?

Comment (Link-) [reply .github/workflows/e2e-test-linux-vm.yaml:13]: Yes, I just saw it's already done that way

File: .github/workflows/e2e-test-linux-vm.yaml (hunk 2 above)
Comment (AvaStancu) [reply .github/workflows/e2e-test-linux-vm.yaml]: For our use case I do not see the need but I am not expert so could be missing something

Comment (TingluoHuang) [reply .github/workflows/e2e-test-linux-vm.yaml]: I think we should include them to make the workflow align with the makefile.

**Review by Link- — APPROVED** [review]
Let's ship it, thank you!

**Pull request merged**